### PR TITLE
[RISCV] Lower f16/bf16 splat_vector by bitcasting to i16 instead of promoting to f32.

### PR DESCRIPTION
**Pull request by topperc**
If f16/bf16 scalar types are not legal we also need to custom legalize to prevent a crash. We do similar lowering for build_vector.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/vsplats-bf16.ll [hunk 1]
```diff
@@ -3,22 +3,58 @@
...
+; CHECK-NEXT:    fmv.x.h a0, fa0
+; CHECK-NEXT:    vsetvli a1, zero, e16, m2, ta, ma
+; CHECK-NEXT:    vmv.v.x v8, a0
+; CHECK-NEXT:    ret
+;
+; NOZFBFMIN-LABEL: vsplat_nxv8bf16:
+; NOZFBFMIN:       # %bb.0:
+; NOZFBFMIN-NEXT:    fmv.x.w a0, fa0
+; NOZFBFMIN-NEXT:    vsetvli a1, zero, e16, m2, ta, ma
+; NOZFBFMIN-NEXT:    vmv.v.x v8, a0
```
Comment (lukel97) [inline llvm/test/CodeGen/RISCV/rvv/vsplats-bf16.ll:23]: How come we have a 32 bit fmv.x.w but then a 16 bit splat?

**Review by lukel97 — COMMENTED** [review]
Should the PR title say "instead of promoting to f32"?

File: llvm/test/CodeGen/RISCV/rvv/vsplats-bf16.ll (hunk 1 above)
Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/vsplats-bf16.ll:23]: Ah I see how we can truncate the bits for bf16.

**Review by lukel97 — APPROVED** [review]

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**